tree/ntuple/v7/inc/ROOT/RPageStorage.hxx

```diff
@@ -367,13 +368,27 @@ protected:
       RNTupleCalcPerf &fFractionReadOverhead;
       RNTupleCalcPerf &fCompressionRatio;
    };
+
+   /// Keeps track of the requested physical column IDs. When using alias columns (projected fields), physical
+   /// columns may be requested multiple times.
+   class RActivePhysicalColumns {
+   private:
+      std::vector<DescriptorId_t> fIDs;
+      std::vector<std::size_t> fRefCounters;
```

Comment:
Thinking about it... the reference counters vector is aligned with the `fIDs` vector, so if `fIDs` becomes an unordered set, `fRefCounters` needs to become an unordered map. We anyway need to copy `fIDs`.  So we would potentially safe time because a `std::unordered_map` copy may be more efficient than constructing it from scratch from a vector (not sure about that). If I'm not missing something, I think I'd slightly prefer the vectors for their simplicity.

Reply:
For now, let's leave it as a `std::vector` then :+1:.